DataFormats/Detectors/Common/include/DetectorsCommonDataFormats/EncodedBlocks.h

```diff
@@ -0,0 +1,616 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file EncodedBlock.h
+/// \brief Set of entropy-encoded blocks
+
+///  Used to store a CTF of particular detector. Can be build as a flat buffer which can be directly messaged between DPL devices
```

Comment:
Basically this is writing a custom allocator for shared memory taken from a pool, right? I don't understand some of the concepts used here. Can you help me understand?

Reply:
sure, EncodedBlocks is a fixed size skeleton which is reinterpret_casted in the beginning of externally provided memory region. It contains fixed amount of block, each one made of the size indicator and a pointer + the pointer on the common registry. The blocks are filled in their index increasing order: the payload of size Block::nStored elements is written in contiguously in the beginning of free memory region, which is registered as a Block::payload pointer, then the free memory is shrunk by changing Registry::offsFreeStart.
If you have more questions, perhaps better to speak on skype.

Reply:
That was already very helpful thank you. If you had time, skype would be even better.